pkg/handlers/lifecycle/cni/multus/template.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package multus
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// templateValuesFunc returns a template function that parses the Multus values template
+// and replaces socket-related placeholders with the provided socket path.
+func templateValuesFunc(socketPath string) func(*clusterv1.Cluster, string) (string, error) {
```

Comment:
How about we keep all the socket related logic in this function and pass the variable instead and then have this function look up the socket?
```suggestion
func templateValuesFunc(cniVar v1alpha1.CNI) func(*clusterv1.Cluster, string) (string, error) {
```